src/survey/ia-feedback-survey.ts

```diff
@@ -0,0 +1,843 @@
+
+    const upvoteColorSvgFilter = css`var(--upvoteColorSvgFilter, invert(34%) sepia(72%) saturate(357%) hue-rotate(111deg) brightness(97%) contrast(95%))`;
+
+    return css`
+      :host {
+        counter-reset: questions;
+      }
+
+      ::slotted(:not(:first-child)) {
+        --surveyQuestionMargin: 15px 0;
+      }
+
+      #container {
+        display: inline-block;
+      }
+
+      #beta-button {
+        font-size: 12px;
+        font-weight: bold;
+        font-style: italic;
+        color: ${betaButtonTextColor};
+        border: 1px solid ${betaButtonBorderColor};
+        border-radius: 4px;
+        padding: 1px 5px;
+      }
+
+      .beta-button-icon {
+        line-height: 100%;
+      }
+
+      .beta-button-icon svg {
+        height: 10px;
+        width: 10px;
+        filter: ${betaButtonSvgFilter};
+      }
+
+      .beta-button-icon.success {
+        filter: ${upvoteColorSvgFilter};
+      }
+
+      #error {
+        margin-bottom: 10px;
+        color: red;
+        font-size: 14px;
+        text-align: center;
+        font-weight: bold;
+      }
+
+      #popup-background {
+        position: fixed;
+        top: 0;
+        left: 0;
+        width: 100%;
+        height: 100%;
+        z-index: ${backdropZindex};
+        background-color: ${popupBlockerColor};
+        overflow: hidden;
+      }
+
+      #popup-background.closed {
+        visibility: hidden;
+        top: -100%;
+        left: -100%;
+      }
+
+      #popup {
+        position: absolute;
+        max-width: ${popupMaxWidth};
+        max-height: calc(100vh - 2 * ${popupVerticalPadding} - 10px);
+        padding: ${popupPadding};
+        border: 1px ${popupBorderColor} solid;
+        border-radius: 5px;
+        background-color: ${popupBackgroundColor};
+        box-shadow: 1px 1px 2px rgba(0, 0, 0, 0.8);
+        z-index: ${modalZindex};
+        margin-left: 10px;
+        margin-right: 10px;
+        overflow-y: auto;
+        scrollbar-width: thin;
+      }
+
+      button,
+      input,
+      a,
+      textarea {
+        font-family: inherit;
+      }
+
+      button,
+      input[type='submit'] {
+        background: none;
+        cursor: pointer;
+        -webkit-appearance: none;
+        -moz-appearance: none;
+        appearance: none;
+        border: none;
+      }
+
+      button:disabled,
+      input[type='submit']:disabled {
+        cursor: default;
+        opacity: 0.5;
+      }
+
+      #form > div:last-child {
+        margin-bottom: 0;
+      }
+
+      #actions {
+        display: flex;
+        justify-content: center;
+      }
+
+      .cta-button {
+        color: white;
+        font-size: 14px;
+        border-radius: 4px;
+        height: 30px;
+        margin: 0;
+      }
+
+      #cancel-button {
+        background-color: ${cancelButtonColor};
+      }
+
+      #submit-button {
+        background-color: ${submitButtonColor};
+        margin-left: 10px;
+      }
+
+      .sr-only,
+      .focus-trap {
```

Comment:
Interesting! What is a focus trap?

Reply:
This is a pattern intended to make the popup act like a modal dialog for the purpose of keyboard navigation. The focus traps are just two invisible, focusable divs at the start and end of the popup which, when focused, immediately pass focus back to the other side of the popup. The end result is that while the popup is open, you can circularly tab through all of its content without keyboard focus ever leaving the popup (until closing it via the Escape key or the Cancel button).

Reply:
Ohh nice! 👍

Reply:
Yeah, agreed!